services/QuillLMS/app/lib/synthetic/result.rb

```diff
@@ -0,0 +1,39 @@
+# frozen_string_literal: true
+
+module Synthetic
+  Result = Struct.new(:text, :label, :type, :generated, keyword_init: true) do
```

Comment:
What made you choose Struct here over a class?   (more just curious since I rarely use structs)

Reply:
I typically use structs when I want a small object to hold a few pieces of related data to make the code easier to read (rather than using a hash or array). I tend to use Structs in another class ([this started as an inline definition](https://github.com/empirical-org/Empirical-Core/pull/9385/commits/f2acec59569d6baaaad9afe7919c572a868da639#diff-f2846d4075ac97b28723f9559ab6dc200c74c0e715d0b7b2a87b3c2df86e02e4R8)) as a handy syntactical sugar to define that object. I added methods later and moved to a separate file, so I could've re-written it as class, just decided not to.

For example, these are more-or-less equivalent:
```ruby
Item = Struct.new(:field1, :field2, keyword_init: true)
# same as
class Item
  attr_accessor :field1, :field2

  def initialize(field1: nil, field2: nil)
    @field1 = field1
    @field2 = field2
  end
end

Reply:
I like the inline definitions within classes.  I kind of wish classes had some sugar to remove the initialize attr_* boiler.plate